src/overlays/actors/ovl_Obj_Takaraya_Wall/z_obj_takaraya_wall.c

```diff
@@ -30,39 +50,450 @@ ActorInit Obj_Takaraya_Wall_InitVars = {
+                                          0x4000, this->actor.shape.rot.z, this->actor.params, this->actor.csId,
+                                          HALFDAYBIT_ALL, NULL);
+    Flags_SetSwitch(play, TAKARAYA_WALL_GET_SWTICH_FLAG(&this->actor));
+    this->actor.shape.rot.z = 0;
+
+    if (chest != NULL) {
+        chest->uncullZoneForward = 2000.0f;
+    }
+
+    sTakarayaWallWallHeights[0][column] = -10.0f;
+    sTakarayaWallPathBuilder[sTakarayaWallPathBuilderIndex].row = 0;
+    sTakarayaWallPathBuilder[sTakarayaWallPathBuilderIndex].column = column;
+    sTakarayaWallPathBuilderIndex++;
+    D_80ADA508++;
+
+    do {
+        ObjTakarayaWall_BuildPath(sTakarayaWallPathBuilder[sTakarayaWallPathBuilderIndex - 1].row,
+                                  sTakarayaWallPathBuilder[sTakarayaWallPathBuilderIndex - 1].column);
+    } while (sTakarayaWallPathBuilderIndex >= 2);
+
+    for (j = 1; j < TAKARAYA_WALL_COLUMNS - 1; j++) {
+        if (sTakarayaWallWallHeights[TAKARAYA_WALL_ROWS - 1][j] == -10.0f) {
+            break;
+        }
+    }
+
+    // If either corner in the front is part of the success path, and there is no other opening in the front,
+    // open up the space next to the open corner since the Treasure Chest Shop scene blocks direct access to the
+    // front corners.
+    if (j == (TAKARAYA_WALL_COLUMNS - 1)) {
+        if (sTakarayaWallWallHeights[TAKARAYA_WALL_ROWS - 1][0] == -10.0f) {
+            sTakarayaWallWallHeights[TAKARAYA_WALL_ROWS - 1][1] = -10.0f;
+        } else {
+            sTakarayaWallWallHeights[TAKARAYA_WALL_ROWS - 1][TAKARAYA_WALL_COLUMNS - 2] = -10.0f;
+        }
+    }
+
+    this->actionFunc = ObjTakarayaWall_Manage;
+}
+
+void ObjTakarayaWall_Destroy(Actor* thisx, PlayState* play) {
+    ObjTakarayaWall* this = THIS;
+    s32 i;
+    s32 j;
+
+    Collider_DestroyCylinder(play, &this->columnCollider);
+    Collider_DestroyCylinder(play, &this->rowCollider);
+
+    for (i = 0; i < TAKARAYA_WALL_ROWS; i++) {
+        for (j = 0; j < TAKARAYA_WALL_COLUMNS; j++) {
+            AudioSfx_StopByPos(&sTakarayaWallAudioPositions[i][j]);
+        }
+    }
+}
+
+void ObjTakarayaWall_Manage(ObjTakarayaWall* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    ColliderCylinder* collider;
+    s32 playerRowBehind;
+    s32 playerRow;
+    s32 playerColumn;
+    s32 columnColliderActive;
+    s32 rowColliderActive;
+    s32 playerSpaceXPos;
+    s32 playerSpaceZPos;
+    s32 i;
+    s32 j;
+    s32 playerColumnRight;
+    s32 playerColumnLeft;
+    s32 playerRowInFront;
+
+    rowColliderActive = false;
+    columnColliderActive = false;
+    playerRowBehind = playerRowInFront = -1;
+    playerColumnLeft = playerColumnRight = -1;
+
+    ObjTakarayaWall_PosToSpace(&player->actor.world.pos, &playerRow, &playerColumn);
+    playerSpaceXPos = (((s32)player->actor.world.pos.x + (15 * 120)) % 120) - 60;
+    playerSpaceZPos = (((s32)player->actor.world.pos.z + (1 * 120)) % 120) - 60;
```

Comment:
```suggestion
    playerSpaceXPos = (((s32)player->actor.world.pos.x - -1800) % 120) - 60;
    playerSpaceZPos = (((s32)player->actor.world.pos.z - -120) % 120) - 60;
```

more world space coordinates